internal/database/sqlcommon/event_sql.go

```diff
@@ -49,6 +49,14 @@ func (s *SQLCommon) InsertEvent(ctx context.Context, event *fftypes.Event) (err
 	}
 	defer s.rollbackTx(ctx, tx, autoCommit)
 
+	// This is special to events - we take the cost of a full table lock on the events table, so
+	// that nobody can add rows that increment the sequence, until our transaction has committed.
+	// This allows us to rely on the sequence to always be increasing, even when writing events
+	// concurrently (it does not guarantee we won't get a gap in the sequences).
```

Comment:
Wonder if it's worth noting this in the database `plugin.go` as well, since we are now requiring any database plugin to provide a stable commit ordering for `InsertEvent` specifically.